src/MarketPlace.sol

```diff
@@ -165,39 +152,22 @@ contract MarketPlace {
         );
 
         {
-            // create the principal tokens array
-            address[9] memory market = [
-                illuminateToken, // Illuminate
-                t[0], // Swivel
-                t[1], // Yield
-                t[2], // Element
-                t[3], // Pendle
-                t[4], // Tempus
-                t[5], // Sense
-                t[6], // APWine
-                t[7] // Notional
-            ];
-
-            // Set the market
-            markets[u][m] = market;
-
-            // Have the lender contract approve the several contracts
-            ILender(lender).approve(u, a, e, t[7], sensePeriphery);
```

Comment:
I see approvals are removed from the Lender -> redeemer in here, does this imply PT custody is on the adapters?

Reply:
Ahh oh yeah the adapter does this approval somehow, so maybe we do need to include approvals still here, but with a passthrough method (with the needed addresses) to the adapter to have it approve properly.

Reply:
That makes sense, and yes, this was a consideration. I added a ticket to get to the approve method for each adapter.